src/types/num.rs

```diff
@@ -10,3 +13,299 @@ use crate::{ffi, PyAny};
+    fn value(&self) -> c_long {
+        unsafe {
+            // Safety: self is PyLong object
+            ffi::PyLong_AsLong(self.as_ptr())
+        }
+    }
+}
+
+/// Implement <Bound<'_, PyLong>> == i8 comparisons
+impl PartialEq<i8> for Bound<'_, PyLong> {
+    #[inline]
+    fn eq(&self, other: &i8) -> bool {
+        self.value() == *other as c_long
+    }
+}
+
+/// Implement i8 == <Bound<'_, PyLong>> comparisons
+impl PartialEq<Bound<'_, PyLong>> for i8 {
+    #[inline]
+    fn eq(&self, other: &Bound<'_, PyLong>) -> bool {
+        *self as c_long == other.value()
+    }
+}
+
+/// Implement <Bound<'_, PyLong>> == u8 comparisons
+impl PartialEq<u8> for Bound<'_, PyLong> {
+    #[inline]
+    fn eq(&self, other: &u8) -> bool {
+        self.value() == *other as c_long
+    }
+}
+
+/// Implement u8 == <Bound<'_, PyLong>> comparisons
+impl PartialEq<Bound<'_, PyLong>> for u8 {
+    #[inline]
+    fn eq(&self, other: &Bound<'_, PyLong>) -> bool {
+        *self as c_long == other.value()
+    }
+}
+
+/// Implement <Bound<'_, PyLong>> == i16 comparisons
+impl PartialEq<i16> for Bound<'_, PyLong> {
+    #[inline]
+    fn eq(&self, other: &i16) -> bool {
+        self.value() == *other as c_long
+    }
+}
+
+/// Implement i16 == <Bound<'_, PyLong>> comparisons
+impl PartialEq<Bound<'_, PyLong>> for i16 {
+    #[inline]
+    fn eq(&self, other: &Bound<'_, PyLong>) -> bool {
+        *self as c_long == other.value()
+    }
+}
+
+/// Implement <Bound<'_, PyLong>> == u16 comparisons
+impl PartialEq<u16> for Bound<'_, PyLong> {
+    #[inline]
+    fn eq(&self, other: &u16) -> bool {
+        self.value() == *other as c_long
+    }
+}
+
+/// Implement u16 == <Bound<'_, PyLong>> comparisons
+impl PartialEq<Bound<'_, PyLong>> for u16 {
+    #[inline]
+    fn eq(&self, other: &Bound<'_, PyLong>) -> bool {
+        *self as c_long == other.value()
+    }
+}
+
+/// Implement <Bound<'_, PyLong>> == i32 comparisons
+impl PartialEq<i32> for Bound<'_, PyLong> {
+    #[inline]
+    fn eq(&self, other: &i32) -> bool {
+        self.value() == *other as c_long
+    }
+}
+
+/// Implement i32 == <Bound<'_, PyLong>> comparisons
+impl PartialEq<Bound<'_, PyLong>> for i32 {
+    #[inline]
+    fn eq(&self, other: &Bound<'_, PyLong>) -> bool {
+        *self as c_long == other.value()
+    }
+}
+
+/// Implement <Bound<'_, PyLong>> == u32 comparisons
+impl PartialEq<u32> for Bound<'_, PyLong> {
+    #[inline]
+    fn eq(&self, other: &u32) -> bool {
+        self.value() == *other as c_long
+    }
+}
+
+/// Implement u32 == <Bound<'_, PyLong>> comparisons
+impl PartialEq<Bound<'_, PyLong>> for u32 {
+    #[inline]
+    fn eq(&self, other: &Bound<'_, PyLong>) -> bool {
+        *self as c_long == other.value()
+    }
+}
+
+/// Implement <Bound<'_, PyLong>> == i64 comparisons
+impl PartialEq<i64> for Bound<'_, PyLong> {
+    #[inline]
+    fn eq(&self, other: &i64) -> bool {
+        self.value() == *other as c_long
+    }
```

Comment:
This and all the following methods will have different behavior on Windows and not Windows systems (because the size of `c_long` is different)

Reply:
Yes, definitely this cast wasn't safe for overflows. Improved to use `.extract()` instead